src/screens/ui/AppletsScreen.tsx

```diff
@@ -41,6 +50,19 @@ const AppletsScreen: FC = () => {
     );
   });
 
+  AppletModel.useEntitiesProgressSync(
```

Comment:
It seems we'll get double work for notification building. Also log files will be uploaded before or in parallel with this logic.
I'd propose to integrate in into existing pipelines: refresh -> [your logic] -> notification.build -> logger.send
See my PR how it done for logger.
The problem is that the pipeline is duplicated in some places. But it's better than to rely on isRefreshing & eventsDownloaded